designs/instance-types-pricing.md

```diff
@@ -0,0 +1,88 @@
+## Design Document for Karpenter: Instance Pricing Override via ConfigMap
+
+### Motivation
+
+Karpenter, an open-source project by AWS, is designed to dynamically provision the right compute resources for Kubernetes workloads. Currently, Karpenter selects instance types based on various factors such as performance, availability, and cost. However, users may need to override the default pricing information to reflect custom pricing agreements, on-premises costs, or special discounts. This feature aims to provide flexibility and precision in cost management by allowing users to specify custom pricing per instance type via a ConfigMap.
+
+### Proposed Specification
+
+#### Overview
+
+Introduce a mechanism to override default instance pricing in Karpenter using a Kubernetes ConfigMap. This ConfigMap will allow administrators to specify custom pricing for various instance types, ensuring Karpenter's scheduling and scaling decisions align with the organization's cost structure.
+
+#### ConfigMap Structure
+
+The ConfigMap will contain key-value pairs where the key represents the instance type and the value represents the custom price. The structure of the ConfigMap will be as follows:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: custom-instance-pricing
+  namespace: karpenter
+data:
+  t3.micro:
+    price: 0.0092
+  m5.large:
+    price: 0.096
+  c5.xlarge:
+    price: 0.17
+```
+
+#### Alternative ConfigMap Structure
+
+The ConfigMap will contain nested key-value pairs where the top-level key represents the instance family, cpu generation and the nested keys represent the instance types with their custom prices. The structure of the ConfigMap will be as follows:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: custom-instance-pricing
+  namespace: karpenter
+data:
+  t:
```

Comment:
How does this kind of format work for things that are in the format `c6in.xlarge`? Does it make more sense to just do a raw list of instance types with keys "t3.micro", "m5.large", etc. and just remove th indentation?